internal/theme/theme.go

```diff
@@ -0,0 +1,13 @@
+package theme
+
+type ColorMode int
+
+const (
+	ThemeLight ColorMode = iota
+	ThemeDark
+	ThemeUnknown
+)
+
+func DetectSystemTheme() ColorMode {
+	return detectSystemTheme() // implemented in per-OS files
```

Comment:
This would fail with other platforms like browsers. Add `theme_others.go` with build tags `//go:build !darwin && !linux && !windows`.